idx/cassandra/config.go

```diff
@@ -0,0 +1,120 @@
+package cassandra
+
+import (
+	"errors"
+	"flag"
+	"time"
+
+	"github.com/rakyll/globalconf"
+	log "github.com/sirupsen/logrus"
+)
+
+// time units accepted by time.ParseDuration
+const timeUnits = "Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'"
+
+// CliConfig is a cassandra IdxConfig. It is instantiated with default values which can then be changed.
+var CliConfig = NewIdxConfig()
+
+// IdxConfig stores configuration settings for a cassandra index
+type IdxConfig struct {
+	Enabled          bool
+	pruneInterval    time.Duration
+	updateCassIdx    bool
+	updateInterval   time.Duration
+	updateInterval32 uint32
+
+	writeQueueSize int
+
+	ssl                      bool
+	auth                     bool
+	hostverification         bool
+	createKeyspace           bool
+	schemaFile               string
+	keyspace                 string
+	hosts                    string
+	capath                   string
+	username                 string
+	password                 string
+	consistency              string
+	timeout                  time.Duration
+	numConns                 int
+	protoVer                 int
+	disableInitialHostLookup bool
+}
+
+// NewIdxConfig returns IdxConfig with default values set.
+func NewIdxConfig() *IdxConfig {
+	return &IdxConfig{
+		Enabled:                  true,
+		hosts:                    "localhost:9042",
+		keyspace:                 "metrictank",
+		consistency:              "one",
+		timeout:                  time.Second,
+		numConns:                 10,
+		writeQueueSize:           100000,
+		updateCassIdx:            true,
+		updateInterval:           time.Hour * 3,
+		pruneInterval:            time.Hour * 3,
+		protoVer:                 4,
+		createKeyspace:           true,
+		schemaFile:               "/etc/metrictank/schema-idx-cassandra.toml",
+		disableInitialHostLookup: false,
+		ssl:                      false,
+		capath:                   "/etc/metrictank/ca.pem",
+		hostverification:         true,
+		auth:                     false,
+		username:                 "cassandra",
+		password:                 "cassandra",
+	}
+}
+
+// Validate validates IdxConfig settings
+func (cfg *IdxConfig) Validate() error {
+	if cfg.updateInterval == 0 {
+		return errors.New("updateInterval must be greater than 0. " + timeUnits)
+	}
```

Comment:
if we're no longer allowing 0 for this value:
1) the help message must be updated
2) this is a breaking change and should be made much more explicit. please mention breaking changes clearly in the PR message.

We can't implement proper validation without stopping supporting a zero value? that seems rather sad.

Reply:
I just tried out `github.com/grafana/globalconf`. We can still support zero values with it.